packages/netsuite-adapter/src/report_definition_parsing/report_definition_parser.ts

```diff
@@ -0,0 +1,124 @@
+/*
+*                      Copyright 2022 Salto Labs Ltd.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with
+* the License.  You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+/* eslint-disable no-underscore-dangle */
+import { Value, Values } from '@salto-io/adapter-api'
+import { collections } from '@salto-io/lowerdash'
+import { ElementCompact } from 'xml-js'
+import {
+  getSearchDependency,
+  getJson,
+  ElementParts,
+  getFlags,
+  extractSearchRecordsValues,
+  safeAssignKeyValue,
+  RecordObject,
+  getObjectFromValues,
+  AttributeObject,
+} from '../saved_search_parsing/saved_search_parser'
+
+const LAYOUTS = 'layouts'
+const PARAMETERS = 'parameters'
+const COMPONENTS = 'components'
+const SORTS = 'sorts'
+const UI_PREFERENCES = 'ui_preferences'
+const FIELDS = 'fields'
+const CRITERIA = 'criteria'
+
+
+type InnerParamObject =
+{
+  _attributes:{
+    class?: string
+    clazz?: string
+    escapeStringIfQuoting?: string
+    field?: string
+    fieldType?: string
+    quoteByNamingConvention?: string
+}
+  _text: string
+}
+type ReportCriterionParent = { _attributs: { class:string; reference: string }}
+type ReportCriterionDescriptor =
+{
+ components: InnerParamObject
+ field:
+ {
+   parent: ReportCriterionParent
+   recordType: InnerParamObject
+   values: { Value: AttributeObject[] }
+  }
+ parent: ReportCriterionParent
+}
+type ReportCriterionValues = { _attributs: { class: string }; Record: RecordObject }
+type ReportCriterionObject = {
+ descriptor: ReportCriterionDescriptor
+ parent: ReportCriterionParent
+ values: ReportCriterionValues
+}
+type ReportCriteria =
+{
+  type: InnerParamObject
+  values:
+  {
+    ReportCriterion: ReportCriterionObject[]
+  }
+}
+type ParameterObject =
+{
+  key: InnerParamObject
+  parent: { class: string; reference: string}
+  value: InnerParamObject
+}
+
+const getReportDefinition = (search: ElementCompact): ElementCompact =>
+  search['nssoc:SerializedObjectContainer']['nssoc:definition'].ReportDefinition
+
+const getReportPartsFromDefinition = async (definition: string): Promise<ElementParts> => {
+  const parsedXml = await getJson(definition)
+  return { definition: getReportDefinition(parsedXml),
+    dependency: getSearchDependency(parsedXml) }
```

Comment:
Styling super nit:
1.
```suggestion
  return {
    definition: getReportDefinition(parsedXml),
    dependency: getSearchDependency(parsedXml),
  }
```
2.
```suggestion
  return { definition: getReportDefinition(parsedXml), dependency: getSearchDependency(parsedXml) }
```